tests/Tokenizer/Transformer/DisjunctiveNormalFormTypeParenthesisTransformerTest.php

```diff
@@ -0,0 +1,290 @@
+                34 => CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_CLOSE,
+                52 => CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_OPEN, // Foo3
+                56 => CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_CLOSE,
+                74 => CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_OPEN, // Foo4
+                78 => CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_CLOSE,
+                80 => CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_OPEN,
+                89 => CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_CLOSE,
+            ],
+        ];
+
+        yield 'multiple functions including methods' => [
+            '<?php
+function Foo():M|(A&B)|J {}
+$a = function(): X|(A&B) {};
+interface Bar {
+    function G():(A&C)|(A&B);
+}
+class G {
+    public function X():K|(Z&Y)|P{}
+}
+',
+            [
+                9 => CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_OPEN,
+                13 => CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_CLOSE,
+                31 => CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_OPEN,
+                35 => CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_CLOSE,
+                53 => CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_OPEN,
+                57 => CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_CLOSE,
+                59 => CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_OPEN,
+                63 => CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_CLOSE,
+                84 => CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_OPEN,
+                88 => CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_CLOSE,
+            ],
+        ];
+
+        yield 'multiple functions including methods 2' => [
+            '<?php
+function Foo():(A&B)|M {}
+$a = function(): (A&B)|N {};
+interface Bar {
+    function G():(A&C)|I;
+}
+class G {
+    public function X():(Z&Y)|P{}
+}
+',
+            [
+                7 => CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_OPEN,
+                11 => CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_CLOSE,
+                27 => CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_OPEN,
+                31 => CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_CLOSE,
+                51 => CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_OPEN,
+                55 => CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_CLOSE,
+                76 => CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_OPEN,
+                80 => CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_CLOSE,
+            ],
+        ];
+
+        yield 'DNF property, A|(C&D)' => [
+            '<?php
+class Dnf
+{
+    public A|(C&D) $a;
+}
+
+const E = 1;
+const F = 2;
+
+(E&F);
+
+',
+            [
+                11 => CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_OPEN,
+                15 => CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_CLOSE,
+            ],
+        ];
+
+        yield 'DNF property, (E&F)|G' => [
+            '<?php
+class Dnf
+{
+    public A|(C&D) $a;
```

Comment:
Shouldn't it be like this?
```suggestion
    public (E&F)|G $a;
```